pkg/executor/sortexec/parallel_sort_spill_helper.go

```diff
@@ -196,17 +207,22 @@ func (p *parallelSortSpillHelper) spillImpl(merger *multiWayMerger) error {
 	go func() {
 		defer func() {
 			if r := recover(); r != nil {
-				processPanicAndLog(p.errOutputChan, r)
+				err = util.GetRecoverError(r)
```

Comment:
This "err" is also used below, which might cause a data race.

Reply:
> This "err" is also used below, which might cause a data race.

fixed